src/test/java/jenkins/plugins/git/GitToolChooserTest.java

```diff
@@ -300,6 +300,36 @@ public void testRemoteAlternatives() throws Exception {
         }
     }
 
+    /* Test conversion of any remote alternative of git repo URLs to a standard URL */
+    @Test
+    public void testConvertToCanonicalURL() throws Exception {
+        GitTool tool = new JGitTool(Collections.<ToolProperty<?>>emptyList());
+
+        String[] remoteAlternatives = {
+                "git://github.com/jenkinsci/git-plugin",
+                "git://github.com/jenkinsci/git-plugin.git",
+                "git@github.com:jenkinsci/git-plugin",
+                "git@github.com:jenkinsci/git-plugin.git",
+                "https://github.com/jenkinsci/git-plugin",
+                "https://github.com/jenkinsci/git-plugin.git",
+                "ssh://git@github.com/jenkinsci/git-plugin",
+                "ssh://git@github.com/jenkinsci/git-plugin.git",
+        };
+
+        String actualNormalizedURL = "https://github.com/jenkinsci/git-plugin.git";
+
+        for (String remote : remoteAlternatives) {
+            GitToolChooser sizeEstimator = new GitToolChooser(remote, null, null, tool, null, TaskListener.NULL, random.nextBoolean());
+            String expectedNormalizedURL = sizeEstimator.convertToCanonicalURL(remote);
+            assertThat("Remote: " + remote, expectedNormalizedURL, is(actualNormalizedURL));
+        }
+
+        /* Check behavior in case of any other format of git repo URL*/
+        String otherRemote = "file://srv/git/repo";
+        GitToolChooser sizeEstimator = new GitToolChooser(otherRemote, null, null, tool, null, TaskListener.NULL, random.nextBoolean());
+        assertThat("Remote: " + otherRemote, sizeEstimator.convertToCanonicalURL(otherRemote), is(otherRemote + ".git"));
```

Comment:
@MarkEWaite Notice how the `convertToCanonicalURL` method converts the `file://srv/git/repo` to `file://srv/git/repo.git`. Since we are storing the normalised URL as a key for the internal cache, I hope this modification is fine.

Reply:
Yes, that is fine as far as I can tell.  The normalized form of the URL needs to be used to access the cache of repository sizes but does not need to be used by any other repository definition in the system.  So long as all cache access goes through the normalization process, we should get correct results from the cache.